server/schedule/region_scatterer.go

```diff
@@ -30,46 +33,69 @@ import (
 )
 
 const regionScatterName = "region-scatter"
+const gcInterval = time.Minute
+const gcTTL = time.Minute * 3
 
 type selectedStores struct {
 	mu sync.Mutex
 	// If checkExist is true, after each putting operation, an entry with the key constructed by group and storeID would be put
 	// into "stores" map. And the entry with the same key (storeID, group) couldn't be put before "stores" being reset
 	checkExist bool
-	// TODO: support auto-gc for the stores
-	stores map[string]map[uint64]struct{} // group -> StoreID -> struct{}
-	// TODO: support auto-gc for the groupDistribution
-	groupDistribution map[string]map[uint64]uint64 // group -> StoreID -> count
+
+	stores            *cache.TTLString // value type: map[uint64]struct{}, group -> StoreID -> struct{}
+	groupDistribution *cache.TTLString // value type: map[uint64]uint64, group -> StoreID -> count
 }
 
-func newSelectedStores(checkExist bool) *selectedStores {
+func newSelectedStores(ctx context.Context, checkExist bool) *selectedStores {
 	return &selectedStores{
 		checkExist:        checkExist,
-		stores:            make(map[string]map[uint64]struct{}),
-		groupDistribution: make(map[string]map[uint64]uint64),
+		stores:            cache.NewStringTTL(ctx, gcInterval, gcTTL),
+		groupDistribution: cache.NewStringTTL(ctx, gcInterval, gcTTL),
 	}
 }
```

Comment:
I think we can directly new a context in the `newSelectedStores` instead of passing it.

Reply:
> I think we can directly new a context in the `newSelectedStores` instead of passing it.

That's my origin idea, but it will cause the ttl's gc goroutine leak, the problem is we do not know when to cancel this context.

I think context is kinda like lifespan for goroutines, and it's reasonable for the ttl's gc goroutine to have the same lifespan with `RegionScatterer`.